pkg/controller/jobs/rayjob/rayjob_controller.go

```diff
@@ -0,0 +1,282 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rayjob
+
+import (
+	"context"
+	"strings"
+
+	rayjobapi "github.com/ray-project/kuberay/ray-operator/apis/ray/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+)
+
+var (
+	gvk = schema.GroupVersionKind{
+		Group:   rayjobapi.GroupVersion.Group,
+		Version: rayjobapi.GroupVersion.Version,
+		Kind:    "RayJob",
+	}
+
+	FrameworkName = "ray.io/rayjob"
+)
+
+func init() {
+	utilruntime.Must(jobframework.RegisterIntegration(FrameworkName, jobframework.IntegrationCallbacks{
+		SetupIndexes:  SetupIndexes,
+		NewReconciler: NewReconciler,
+		SetupWebhook:  SetupRayJobWebhook,
+		JobType:       &rayjobapi.RayJob{},
+		AddToScheme:   rayjobapi.AddToScheme,
+	}))
+}
+
+// RayJobReconciler reconciles a Job object
+type RayJobReconciler jobframework.JobReconciler
+
+func NewReconciler(
+	scheme *runtime.Scheme,
+	client client.Client,
+	record record.EventRecorder,
+	opts ...jobframework.Option) jobframework.JobReconcilerInterface {
+	return (*RayJobReconciler)(jobframework.NewReconciler(scheme,
+		client,
+		record,
+		opts...,
+	))
+}
+
+type RayJob rayjobapi.RayJob
+
+func (j *RayJob) Object() client.Object {
+	return (*rayjobapi.RayJob)(j)
+}
```

Comment:
```suggestion
type RayJob struct {
	*rayjobapi.RayJob
}

func (j *RayJob) Object() client.Object {
	return j.RayJob
}
```

Avoiding casting like this would be good.  If we do this, we can avoid casting in webhook.

Reply:
I don't understand.  If we use the embedded  pointer version, we have an extra unnecessary allocation every time we instantiate a `kueue. RayJob`.